Content.Server/Anomaly/AnomalySystem.Scanner.cs

```diff
@@ -132,29 +132,95 @@ public FormattedMessage GetScannerMessage(AnomalyScannerComponent component)
             return msg;
         }
 
-        msg.AddMarkup(Loc.GetString("anomaly-scanner-severity-percentage", ("percent", anomalyComp.Severity.ToString("P"))));
+        TryComp<SecretDataAnomalyComponent>(anomaly, out var secret);
+
+        //Severity
+        if (secret != null && secret.Secret.Contains(AnomalySecretData.Severity))
+            msg.AddMarkup(Loc.GetString("anomaly-scanner-severity-percentage-unknown"));
+        else
+            msg.AddMarkup(Loc.GetString("anomaly-scanner-severity-percentage", ("percent", anomalyComp.Severity.ToString("P"))));
         msg.PushNewline();
-        string stateLoc;
-        if (anomalyComp.Stability < anomalyComp.DecayThreshold)
-            stateLoc = Loc.GetString("anomaly-scanner-stability-low");
-        else if (anomalyComp.Stability > anomalyComp.GrowthThreshold)
-            stateLoc =  Loc.GetString("anomaly-scanner-stability-high");
+
+        //Stability
+        if (secret != null && secret.Secret.Contains(AnomalySecretData.Stability))
+            msg.AddMarkup(Loc.GetString("anomaly-scanner-stability-unknown"));
         else
-            stateLoc =  Loc.GetString("anomaly-scanner-stability-medium");
-        msg.AddMarkup(stateLoc);
+        {
+            string stateLoc;
+            if (anomalyComp.Stability < anomalyComp.DecayThreshold)
+                stateLoc = Loc.GetString("anomaly-scanner-stability-low");
+            else if (anomalyComp.Stability > anomalyComp.GrowthThreshold)
+                stateLoc = Loc.GetString("anomaly-scanner-stability-high");
+            else
+                stateLoc = Loc.GetString("anomaly-scanner-stability-medium");
+            msg.AddMarkup(stateLoc);
+        }
         msg.PushNewline();
 
-        msg.AddMarkup(Loc.GetString("anomaly-scanner-point-output", ("point", GetAnomalyPointValue(anomaly, anomalyComp))));
+        //Point output
+        if (secret != null && secret.Secret.Contains(AnomalySecretData.OutputPoint))
+            msg.AddMarkup(Loc.GetString("anomaly-scanner-point-output-unknown"));
+        else
+            msg.AddMarkup(Loc.GetString("anomaly-scanner-point-output", ("point", GetAnomalyPointValue(anomaly, anomalyComp))));
         msg.PushNewline();
         msg.PushNewline();
 
+        //Particles title
         msg.AddMarkup(Loc.GetString("anomaly-scanner-particle-readout"));
         msg.PushNewline();
-        msg.AddMarkup(Loc.GetString("anomaly-scanner-particle-danger", ("type", GetParticleLocale(anomalyComp.SeverityParticleType))));
+
+        //Danger
+        if (secret != null && secret.Secret.Contains(AnomalySecretData.ParticleDanger))
```

Comment:
bruuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuh

make these components

Reply:
![image](https://github.com/space-wizards/space-station-14/assets/96445749/3b532d02-eeb4-45af-943c-2171c4c7a03a)

We have 4 fields that change every time, even if we break this code into a separate function, it will take 4 parameters, and it will not look better

Reply:
-unknown doesnt change
if there is a dict of particle to projectile prototype this can just be 1 var changing less copypasta